### PR TITLE
ramips: Add support for Unitac V-FL500 a.k.a Skylink 4G V-FL500

### PR DESCRIPTION
**Pull request by LynxChaus**
ramips: Add support for Unitac V-FL500 a.k.a Skylink 4G V-FL500

Specification:
- SoC: MediaTek MT7620N
- Flash: 8 MB
- RAM: 64 MB
- Ethernet: 3 FE ports (LLW)
- Wireless radio: MT7620 for 2.4G
- UART: 1 x UART on PCB - 57600 8N1 (3.3v TTL Vcc,Tx,Rx,GND from ethernet ports)
- USB: 1 x USB 2.0 port routed to internal miniPCIe slot with Altair ALT3800 4G LTE modem
- LEDs: 1 - hardwired to Vcc, 1 - SoC GPIO,
  4x LEDs controlled via I2C Allwinc AW9523 GPIO controller in LED-driver mode
- 2x KEYs

Flash instruction:

Login to original firmware (RALINK SDK based on OpenWrt 14.07) and flash wia web.

Signed-off-by: Andrey Jr. Melnikov <temnota.am@gmail.com>

### CODE REVIEW

File: package/kernel/i2c-aw9523-expander/src/i2c-leds-aw9523.c [hunk 1]
```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: GPL-2.0+
```
Comment (mkresin) [inline package/kernel/i2c-aw9523-expander/src/i2c-leds-aw9523.c]: Is this driver upstream to the linux kernel? If not, please do so and backport the driver. We are doing out best to reduce the amount of custom driver/patches.

Comment (LynxChaus) [reply package/kernel/i2c-aw9523-expander/src/i2c-leds-aw9523.c]: I don's see any reason to upstream this stub driver for unknown Taiwan IC found in cheapest Chinese router specially make for Russian 4G LTE operator.

**Comment by mkresin** [comment]
> @LynxChaus commented on this pull request.
> 
> In package/kernel/i2c-aw9523-expander/src/i2c-leds-aw9523.c:
> 
> > @@ -0,0 +1,269 @@
> +// SPDX-License-Identifier: GPL-2.0+
> I don's see any reason to upstream this stub driver for unknown Taiwan IC found in cheapest Chinese router specially make for Russian 4G LTE operator.
> 

And I don't see a reason to burden OpenWrt Developers with the maintenance of a driver for an unknown Taiwan IC found in cheapest Chinese router specially make for Russian 4G LTE operator.

We all know that out of tree kernel driver need constant updates/adjustments to keep track with upstream changes.

So, here we are. How to proceed?

**Comment by mkresin** [comment]
I fixed the formal issues of your "Move i2c_pins definition to right place" commit, merged and backported it.

**Comment by mkresin** [comment]
Please reopen the PR if you have the time to address the requested changes.

**Comment by qmastery16** [comment]
@LynxChaus @mkresin Sad to see this port hasn't been merged. Do you have any chance to continue this work?

File: package/kernel/i2c-aw9523-expander/src/i2c-leds-aw9523.c (hunk 1 above)
Comment (qmastery16) [reply package/kernel/i2c-aw9523-expander/src/i2c-leds-aw9523.c:1]: @mkresin @LynxChaus Looks like aw9523.c driver has been merged into a Linux kernel :smiley: Please go ahead

**Comment by LynxChaus** [comment]
@qmastery16, mainline aw9523 driver is ONLY GPIO. V-FL500 need PWM driver for LED's.
And no, this PR is already dead - V-FL500 have 8Mb flash, and 19.x release not fit to it.

**Pull request closed without merging**